# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientSource.cpp
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/clients/manager/ClientSource.cpp
--- a/dom/clients/manager/ClientSource.cpp	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/clients/manager/ClientSource.cpp	Thu Oct 24 03:02:42 2024 +0000
@@ -163,8 +163,9 @@
     : mManager(aManager),
       mEventTarget(aEventTarget),
       mOwner(AsVariant(Nothing())),
-      mClientInfo(aArgs.id(), aArgs.type(), aArgs.principalInfo(),
-                  aArgs.creationTime()) {
+      mClientInfo(aArgs.id(), aArgs.agentClusterId(), aArgs.type(),
+                  aArgs.principalInfo(), aArgs.creationTime(), aArgs.url(),
+                  aArgs.frameType()) {
   MOZ_ASSERT(mManager);
   MOZ_ASSERT(mEventTarget);
 }
@@ -186,9 +187,10 @@
     return;
   }
 
-  ClientSourceConstructorArgs args(mClientInfo.Id(), mClientInfo.Type(),
-                                   mClientInfo.PrincipalInfo(),
-                                   mClientInfo.CreationTime());
+  ClientSourceConstructorArgs args(
+      mClientInfo.Id(), Nothing(), mClientInfo.Type(),
+      mClientInfo.PrincipalInfo(), mClientInfo.CreationTime(), VoidCString(),
+      FrameType::None);
   RefPtr<ClientSourceChild> actor = new ClientSourceChild(args);
   if (!aActor->SendPClientSourceConstructor(actor, args)) {
     Shutdown();